Source/WebCore/page/SecurityOrigin.cpp

```diff
@@ -57,22 +57,9 @@ bool SecurityOrigin::shouldIgnoreHost(const URL& url)
     return url.protocolIsData() || url.protocolIsAbout() || url.protocolIsJavaScript() || url.protocolIs("file"_s);
 }
 
-bool SecurityOrigin::shouldUseInnerURL(const URL& url)
+URL SecurityOrigin::extractURLFromBlobURLPath(const URL& url)
```

Comment:
This needs at least an ASSERT(url.protocolIsBlob()).
I think it would be safer to move the url.protocolIsBlob() to inside this function instead of requiring all callers to check before calling.